zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/DocumentenApiPlugin.kt

```diff
@@ -65,24 +67,70 @@ class DocumentenApiPlugin(
         @PluginActionProperty localDocumentLocation: String,
         @PluginActionProperty storedDocumentUrl: String,
         @PluginActionProperty informatieobjecttype: String,
-        @PluginActionProperty taal: String = "nld",
+        @PluginActionProperty taal: String = DEFAULT_LANGUAGE,
         @PluginActionProperty status: DocumentStatusType = DocumentStatusType.DEFINITIEF
     ) {
         val documentLocation = execution.getVariable(localDocumentLocation) as String
         val contentAsInputStream = storageService.getResourceContentAsInputStream(documentLocation)
+        val metadata = storageService.getResourceMetadata(documentLocation)
 
         val request = CreateDocumentRequest(
             bronorganisatie = bronorganisatie,
+            creatiedatum = getLocalDateFromMetaData(metadata, "creationDate", LocalDate.now())!!,
             titel = title,
-            vertrouwelijkheidaanduiding = ConfidentialityLevel.fromKey(confidentialityLevel).key,
+            vertrouwelijkheidaanduiding = getConfidentialityLevel(confidentialityLevel),
+            auteur = metadata["author"] as String? ?: DEFAULT_AUTHOR,
             status = status,
             taal = taal,
             bestandsnaam = fileName,
             inhoud = contentAsInputStream,
             beschrijving = description,
+            ontvangstdatum = getLocalDateFromMetaData(metadata, "receiptDate"),
+            verzenddatum = getLocalDateFromMetaData(metadata, "sendDate"),
             informatieobjecttype = informatieobjecttype,
         )
+        storeDocument(execution, request, storedDocumentUrl)
+    }
+
+    @PluginAction(
```

Comment:
This new PluginAction is very similar to the one above. Initially I wanted to combine them. But then all the PluginActionProperty of the plugin above should become optional. Im not sure if thats what we want

Reply:
I don't have a problem with having 2 separate actions, but the code duplication bothers me. You could make it more generic if you'd merge the properties of storeTemporaryDocument with the resource metadata into a new map and pass that to a private method that can be used by both actions.